src/server/stream_family.cc

```diff
@@ -1949,77 +1986,85 @@ OpResult<int64_t> OpTrim(const OpArgs& op_args, const AddTrimOpts& opts) {
   CompactObj& cobj = res_it->it->second;
   stream* s = (stream*)cobj.RObjPtr();
 
-  auto res = StreamTrim(opts, s);
+  auto res = TrimStream(opts, s);
 
   mem_tracker.UpdateStreamSize(cobj);
-  return res;
+
+  if (op_args.shard->journal()) {
+    if (JournalAsMinId(opts)) {
+      // We need to set exact MinId in the journal.
+      std::string last_id = StreamsIdToString(res.second);
+      RecordJournal(op_args, "XTRIM"sv, ArgSlice{key, "MINID"sv, "="sv, last_id});
+    } else {
+      RecordJournal(op_args, "XTRIM"sv, shard_args);
```

Comment:
I cases where the journal does not need rewrite you can use 
ReviveAutoJournal before you call OpTrim
This way you will not need to pass shard_args